imitation/scripts/data_collect.py

```diff
@@ -9,21 +11,112 @@
 
 
 @data_collect_ex.main
-def main(_seed, env_name, log_dir, parallel, total_timesteps,
-         num_vec=8, make_blank_policy_kwargs={}):
-  tf.logging.set_verbosity(tf.logging.INFO)
-  sb_logger.configure(folder=osp.join(log_dir, 'rl'),
-                      format_strs=['tensorboard', 'stdout'])
-
-  env = util.make_vec_env(env_name, num_vec, seed=_seed,
-                          parallel=parallel, log_dir=log_dir)
-  # TODO(adam): add support for wrapping env with VecNormalize
-  # (This is non-trivial since we'd need to make sure it's also applied
-  # when the policy is re-loaded to generate rollouts.)
-  policy = util.make_blank_policy(env, verbose=1, **make_blank_policy_kwargs)
-
-  callback = util.make_save_policy_callback(osp.join(log_dir, 'checkpoints'))
-  policy.learn(total_timesteps, callback=callback)
+def main(_seed: int,
+         env_name: str,
+         total_timesteps: int,
+         *,
+         log_dir: str = None,
+         parallel: bool = False,
+         num_vec: int = 8,
+         make_blank_policy_kwargs: dict = {},
+
+         rollout_save: bool = False,
+         rollout_save_interval: Optional[int] = None,
+         rollout_save_n_samples: int = 2000,
+         rollout_dir: str = "data/rollouts",
+
+         policy_save: bool = False,
+         policy_save_interval: Optional[int] = None,
+         policy_dir: str = "data/policies",
+         ):
+  """Train a policy from scratch, optionally saving the policy and rollouts.
+
+  At applicable training steps `step`,
+    - Policies are saved to
+      `{policies_dir}/{env_name}-{policy_name}-{step}.pkl`.
```

Comment:
If I run `data_collect` twice on the same environment with different random seeds, the policies and rollouts will overwrite each other, unless I change the `policies_dir` and `rollout_dir` explicitly. Also, I have no way of telling what hyperparameters were used to generate a particular policy.

I generally favour treating each invocation of a command as a separate (sub)experiment, and storing all its output in a unique directory. I tend not to delete anything, unless the output is really huge.

The simple solution here is to just scrap `rollout_dir` and `policy_dir` and replace them with `os.path.join(log_dir, 'rollouts')` and `os.path.join(log_dir, 'policies')` (or, set those as defaults). `log_dir` is uniquely generated (with environment name & a timestamp). It's sadly not in the same directory as the Sacred output, but you can at least link them by searching through the Sacred configs for the relevant `log_dir`.